core/src/main/java/org/infinispan/manager/CacheManagerInfo.java

```diff
@@ -186,6 +193,37 @@ public boolean isTracingEnabled() {
                  .getGlobalConfiguration().tracing().enabled();
    }
 
+   public boolean allCachesHealthy() {
+      return cacheManager.getCaches().keySet().stream()
+            .allMatch(this::isCacheReady);
+   }
+
+   public boolean isCacheReady(String cacheName) {
+      try {
+         GlobalComponentRegistry gcr = SecurityActions.getGlobalComponentRegistry(cacheManager);
+         ComponentRegistry cr = gcr.getNamedComponentRegistry(cacheName);
+
+         log.debugf("Verifying %s, CR=%s", cacheName, cr);
```

Comment:
This should be trace I think as this is called every so often even after startup right?

Reply:
Should we also not print a trace based on why it was false?

Reply:
That would definitely help to debug if the cluster starts replying as not ready.